torchtext/vocab/vocab_factory.py

```diff
@@ -34,19 +36,27 @@ def vocab(ordered_dict: Dict, min_freq: int = 1) -> Vocab:
         >>> unk_token = '<unk>'
         >>> default_index = -1
         >>> if unk_token not in v2: v2.insert_token(unk_token, 0)
-        >>> v2.set_default_index(default_index)
-        >>> print(v2['<unk>']) #prints 0
-        >>> print(v2['out of vocab']) #prints -1
+        >>> v3 = vocab(OrderedDict([(token, 1) for token in tokens]), specials=[unk_token])
+        >>> v3.set_default_index(default_index)
+        >>> print(v3['<unk>']) #prints 0
+        >>> print(v3['out of vocab']) #prints -1
         >>> #make default index same as index of unk_token
-        >>> v2.set_default_index(v2[unk_token])
-        >>> v2['out of vocab'] is v2[unk_token] #prints True
+        >>> v3.set_default_index(v3[unk_token])
+        >>> v3['out of vocab'] is v3[unk_token] #prints True
     """
+    for token in specials:
+        ordered_dict.pop(token, None)
 
     tokens = []
     for token, freq in ordered_dict.items():
         if freq >= min_freq:
             tokens.append(token)
 
+    if special_first:
+        tokens[0:0] = specials
```

Comment:
Just out of curiosity, what does `tokens[0:0]` do? I've never seen this syntax before, but I'm assuming this is prepending `specials` to the `tokens` list. Time-complexity wise, is there any benefit to doing this as opposed to list concatenation i.e. `tokens = specials + tokens`?

Reply:
You are right that this syntax is used to prepend specials to the tokens. Time complexity wise there is no difference. However, space-complexity wise this is better because it prepends in-place while list concatenation creates a new list.